app/models/planet.py

```diff
@@ -0,0 +1,37 @@
+from app import db
+from flask import abort, make_response
+
+
+class Planet(db.Model):
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    name = db.Column(db.String)
+    description = db.Column(db.String)
+    association = db.Column(db.String)
+
+    def to_dict(self):
+        return dict(
+            id=self.id,
+            name=self.name,
+            description=self.description,
+            association=self.association,
+            )
+    
+    @classmethod
+    def from_dict(cls, planet_data):
+        try:
+            new_planet = Planet(name=planet_data["name"],
+                                description=planet_data["description"],
+                                association=planet_data["association"],
+                                )
+        except KeyError: 
+            abort(make_response({"message": f"Provided planet data was invalid"}, 400))
+        
+        return new_planet
+
+    def update_from_dict(self, planet_data):
```

Comment:
👍 Nice addition of this helper to update the model values.

Notice that by moving the abort/make_response logic inside this helper (and for the create) it does simplify the logic out in the routes, but it means that our model type now has to take a dependency on functions from Flask, whereas previously it only had a db dependency (the imports we need).

In a larger system, we might resist this change, as we say that it couples our implementation to Flask. Instead, we might introduce a new layer, one that sits at the border between the routes (Flask) and the models (SqlAlchemy), whose responsibility is to bridge that transition. In that configuration, we could keep the logic to read from the dictionary here, but move the `try`/`except` logic to this new intermediate layer. This would keep the route clean, and avoid coupling our model type to Flask.

But of course there are many ways to tackle this. Another option could be to write a custom decorator (we've only been using decorators, but we can write them as well), that we could use to decorate any routes that could result in a KeyError, and translate to the appropriate abort message.

For the scale of these activities, either might be considered overkill, but in larger systems, we like to tease out these responsibilities into their own areas of code.